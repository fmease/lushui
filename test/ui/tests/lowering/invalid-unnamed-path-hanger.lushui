;;; TEST fail check

use super
use crate
use other
use self

use crate.(space self space)

use crate.(self.(self.thing super))

use exists.(self)
