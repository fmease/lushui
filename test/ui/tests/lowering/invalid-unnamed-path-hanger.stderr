error[E025]: path `super` is not bound to an identifier
 --> ${TEST_FOLDER}/lowering/invalid-unnamed-path-hanger.lushui:3:5
  |
3 | use super
  |     ^^^^^ 
  |
 note: a use-declaration has to introduce at least one new binder
 help: bind the path to a name with `as`

error[E025]: path `topmost` is not bound to an identifier
 --> ${TEST_FOLDER}/lowering/invalid-unnamed-path-hanger.lushui:4:5
  |
4 | use topmost
  |     ^^^^^^^ 
  |
 note: a use-declaration has to introduce at least one new binder
 help: bind the path to a name with `as`

error[E025]: path `self` is not bound to an identifier
 --> ${TEST_FOLDER}/lowering/invalid-unnamed-path-hanger.lushui:6:5
  |
6 | use self
  |     ^^^^ 
  |
 note: a use-declaration has to introduce at least one new binder
 help: bind the path to a name with `as`

error[E025]: path `self` is not bound to an identifier
 --> ${TEST_FOLDER}/lowering/invalid-unnamed-path-hanger.lushui:8:20
  |
8 | use topmost.(space self space)
  |                    ^^^^ 
  |
 note: a use-declaration has to introduce at least one new binder
 help: bind the path to a name with `as`

error[E026]: path `super` not allowed in this position
  --> ${TEST_FOLDER}/lowering/invalid-unnamed-path-hanger.lushui:10:31
   |
10 | use topmost.(self.(self.thing super))
   |                               ^^^^^ 
   |
  help: consider moving this path to a separate use-declaration

error: aborting due to 5 previous errors

